lnrpc/walletrpc/walletkit.proto

```diff
@@ -917,6 +917,14 @@ message LabelTransactionRequest {
 message LabelTransactionResponse {
 }
 
+enum KeyScope {
+    KEY_SCOPE_UNSPECIFIED = 0;
+    KEY_SCOPE_BIP0086 = 1;
```

Comment:
Wouldn't using identifiers like `p2tr` be easier for users than bip numbers?

Reply:
Hmm I am not really sure

Because for example `KeyScopeBIP0049Plus` and `KeyScopeBIP0084` will result to a  P2WPKH.

So, without using BIP number, it will be difficult to differentiate them?

Reply:
I thought bip49 is np2wsh (nested). Btw, there is already an `AddressType` enum in `lightning.proto`.

Reply:
Actually:

```
// KeyScopeBIP0049Plus is the key scope of our modified BIP0049
// derivation. We say this is BIP0049 "plus", as we'll actually use
// p2wkh change all change addresses.
```

Otherwise, I would be happy to use address type instead as it would be easier for the user :(

Reply:
So it is for the purpose of the FundPsbt change output the same as using bip84? Wondering if "bip49+" should be an option then for the change scope, because it seems redundant?

Reply:
Yeah, I think you are right.

Let's drop BIP0049Plus and use simpler identifier for the keyscope

Reply:
Can't we just use the existing `AddressType` here? That maps nicely to a key scope (there are even helper functions present for that I think) and is a bit more verbose than BIP0049Plus vs BIP0049 (even though the difference between NESTED and HYBRID_NESTED is probably not obvious either).

Reply:
I do agree that it would be easier to use existing `AddressType`, but I don’t think we can use lightning.proto `AddressType` because of the enum default (which is p2wkh in lightning.proto).

Imagine that the user specifies an account (with p2tr scope) but doesn't specify the key scope for the output. Then by default we will use P2WKH (and we have no way to know if it was because of the default behaviour or not).
And if we decide to change the default address in lightning.proto, we may induce a breaking? (`NewAddress` behaviour will change)

With `KEY_SCOPE_UNSPECIFIED`, we don't have a default key scope. 
IMO, We shouldn’t have a default key scope for the change set by the proto (especially because `changescope`  is optional)

What do you think?

Reply:
(PS: depending on the https://github.com/lightningnetwork/lnd/pull/7348#discussion_r1084013324 answer, this problem will be solved :) )

Reply:
> Otherwise, I would be happy to use address type instead as it would be easier for the user :(

nit: Just saw the typo in the snippet of code you posted `for all change addresses`

Reply:
The snippet I posted is the godoc of BIP0049Plus (if this is what you meant):

https://github.com/btcsuite/btcwallet/blob/master/waddrmgr/scoped_manager.go#L171-L173

Indeed there is a little typo. Maybe I can take advantage of the btcsuite PR to solve this typo

Reply:
>Can't we just use the existing AddressType here? That maps nicely to a key scope (there are even helper functions present for that I think) and is a bit more verbose than BIP0049Plus vs BIP0049 (even though the difference between NESTED and HYBRID_NESTED is probably not obvious either).

Some other thoughts:

First, if we use `AddressType` instead of BIP code, the user will be able (in the rpc call) to provide
`n2wpkh` and `hybrid-n2wpkh` as the `change_address`, and this will be mapped to BIP0049Plus which will use a p2wkh for the change of address (confusing)
(Maybe we can return an error if the user tries to use one of the two, or explain the behaviour in the godoc but...)

Second, if we create a new `ChangeAddressType` struct in `wallet.proto` with only p2tr, p2wkh, p2kh. 
How do we choose the key scope for `p2wkh`? (between BIP0084 and BIP0049Plus, since both use p2wkh as `InternalAddrType`). 
If we imagine that we force BIP0084 when `change_address = p2wkh`.
If a user provides an account with a unique scope (BIP0049Plus) and `p2wkh` for the change address in FundPSBT, then, is it acceptable to send an error because we tried to use BIP0084 even though BIP0049Plus might work?

That’s only thoughts, I am not against the idea 😄